ui/src/main/java/com/epam/eco/kafkamanager/ui/topics/browser/handlers/FilterOperationJsonHandler.java

```diff
@@ -0,0 +1,143 @@
+            JsonNode node = mapper.readTree(value);
+            return compareValue(node,
+                                getFieldName(getClause().getValue()),
+                                getFieldValue(getClause().getValue()),
+                                String::startsWith);
+        } catch (JsonProcessingException e) {
+            return value.startsWith(getClause().getValue());
+        }
+    }
+
+    @Override
+    boolean like(String value) {
+        try {
+            JsonNode node = mapper.readTree(value);
+            return compareValue(node,
+                                getFieldName(getClause().getValue()),
+                                getFieldValue(getClause().getValue()),
+                                FilterOperationUtils::like);
+        } catch (JsonProcessingException e) {
+            return FilterOperationUtils.like(getClause().getValue(), value);
+        }
+    }
+
+    @Override
+    boolean notEmpty(String value) {
+        try {
+            JsonNode node = mapper.readTree(value);
+            return compareValue(node,
+                                getFieldName(getClause().getValue()),
+                                getFieldValue(getClause().getValue()),
+                                FilterOperationUtils::notEmpty);
+        } catch (JsonProcessingException e) {
+            return FilterOperationUtils.like(getClause().getValue(), value);
+        }
+    }
+
+    private String getFieldName(String value) {
+       Validate.notNull(value);
+       return value.substring(0,value.indexOf(KEY_VALUE_SEPARATOR)).trim();
+    }
+    private String getFieldValue(String value) {
+        Validate.notNull(value);
+        return value.substring(value.indexOf(KEY_VALUE_SEPARATOR)+1).trim();
+    }
+
+    private boolean compareValue(JsonNode node,
+                                 String fieldName,
+                                 String testValue,
+                                 BiPredicate<String,String> compareMethod) {
+        if(node.isArray()) {
+            ArrayNode arrayNode = (ArrayNode) node;
+            for(JsonNode singleNode: arrayNode) {
+                boolean result = compareValue(singleNode, fieldName, testValue, compareMethod);
+                if(result) {
+                    return true;
+                }
+            }
+        } else {
+            if(fieldName.contains(PROPERTIES_SEPARATOR)) {
+                String currentFieldName = fieldName.substring(0,fieldName.indexOf(PROPERTIES_SEPARATOR));
+                if(node.has(currentFieldName)) {
+                    String restFieldName = fieldName.substring(fieldName.indexOf(PROPERTIES_SEPARATOR) + 1);
+                    return compareValue(node.get(currentFieldName),restFieldName,testValue,compareMethod);
+                } else {
+                    return false;
+                }
+
+            } else {
+                return node.has(fieldName) && compareMethod.test(node.get(fieldName).asText(), testValue);
```

Comment:
Try to check if **.asText()** is suitable for comparing numbers, floats especially. Or write some tests, maybe

Reply:
I checked it. All things looks good.